docs/DeveloperGuide.adoc

```diff
@@ -279,104 +277,440 @@ Refer to the guide <<DevOps#, here>>.
+
+|`* *`|student|get a statistics of the number of questions I have attempted at a difficulty level|I can see how often I have practiced for this subject|Xueting
+
+|`* *`|student|get statistics of the accuracy level of the specific difficulty|I can visualise how well I have practiced for this subject|Xueting
+
+|`*`|student|get statistics of the change of difficulty levels I have made to a question|I can understand my improvement in understanding the content|Xueting
+
+|`* * *`|student|see how well I am able to answer questions for different topics/subjects|focus more on topics/subjects that I need to improve on|Irene
+
+|`* * *`|student|check my past responses to different questions|compare with my current response and improve on it if necessary/check if I am making the same mistakes twice|Irene
+
+|`* *`|student|have a graphical report of my progress|get an overview of my progress with a glimpse|Irene
+
+|`* *`|student|check the questions that I have attempted for a specific time period|I can skip questions that I have already done during that time period|Irene
+
+|`*`|student|get an overview to see the number of notes added for each topic/subject|I can see which topic/subject I need to make more notes on|Irene
+
+|`* * *`|student|mark and store the questions I have made mistakes on|I can get a better understanding of which part I need to put more time to study|Dongjun
 
-_{More to be added}_
+|`* * *`|student|take a series questions as a mock test before exam|I can refresh my memory on different modules before exam|Dongjun
+
+|`* *`|student|find specific questions according to the keywords|I can filter the questions I want for different purposes|Dongjun
+
+|`* *`|student|change the difficulty of my test|I can be better prepared for actual exam|Dongjun
+
+|`*`|student|view all questions and answers stored|I can refer them as revision guide according to different subjects|Dongjun
+
+|=======================================================================
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the system is `NUStudy` and the actor is the `student`)
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Create lecture note (Jeremy)
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
-+
-Use case ends.
+1. Student gives the command to add a lecture note – `addnote -t TITLE -c CONTENT`
+2. Application shows a lecture note with the given title and contents to the user
+3. Application also updates the lecture note list with the new note
+
+Use case ends
+
+*Extensions*
+
+[none]
+* 1a. Student also gives `-i` in the command
+[none]
+** 1a1. Application brings up a file selection dialogue
```

Comment:
Should probably avoid concrete mentions of UI, can maybe say "application allows user to select file".